transport-native-unix-common/src/main/java/io/netty/channel/unix/Errors.java

```diff
@@ -102,6 +102,34 @@ public synchronized Throwable fillInStackTrace() {
         }
     }
 
+    public static final class NativeCallException extends RuntimeException {
+        private static final long serialVersionUID = -5532328671712318161L;
+        private final int expectedErr;
+        private final boolean fillInStackTrace;
+
+        public NativeCallException(String method, int expectedErr) {
+            this(method, expectedErr, true);
+        }
+
+        public NativeCallException(String method, int expectedErr, boolean fillInStackTrace) {
+            super(method + "(..) failed: " + errnoString(-expectedErr));
+            this.expectedErr = expectedErr;
+            this.fillInStackTrace = fillInStackTrace;
+        }
+
+        public int expectedErr() {
+            return expectedErr;
+        }
+
+        @Override
+        public synchronized Throwable fillInStackTrace() {
```

Comment:
nit: You can remove the `synchronized`
```suggestion
        public Throwable fillInStackTrace() {
```